packages/asset/test/AssetCreate.test.ts

```diff
@@ -1572,4 +1572,576 @@ describe('AssetCreate (/packages/asset/contracts/AssetCreate.sol)', function ()
+        await mintCatalyst(4, 5);
+        const signature = await generateLazyMintSignature(
+          creator.address,
+          4,
+          5,
+          metadataHashes[0],
+          10
+        );
+        const nonceBefore = await AssetCreateContract.creatorNonces(
+          creator.address
+        );
+        await lazyMintAsset(
+          signature,
+          4,
+          5,
+          metadataHashes[0],
+          10,
+          creator.address
+        );
+        const nonceAfter = await AssetCreateContract.creatorNonces(
+          creator.address
+        );
+        expect(nonceAfter).to.be.gt(nonceBefore);
+      });
+      it('should increase the amount for already minted asset', async function () {
+        const {
+          mintCatalyst,
+          lazyMintAsset,
+          generateLazyMintSignature,
+          metadataHashes,
+          creator,
+          AssetContract,
+        } = await runCreateTestSetup();
+        await mintCatalyst(4, 10);
+
+        const firstMintSignature = await generateLazyMintSignature(
+          creator.address,
+          4,
+          5,
+          metadataHashes[0],
+          10
+        );
+        const firstMintResult = await lazyMintAsset(
+          firstMintSignature,
+          4,
+          5,
+          metadataHashes[0],
+          10,
+          creator.address
+        );
+
+        const firstEventData = firstMintResult.events.filter(
+          (e: Event) => e.event == 'AssetLazyMinted'
+        )[0].args;
+
+        const secondMintSignature = await generateLazyMintSignature(
+          creator.address,
+          4,
+          5,
+          metadataHashes[0],
+          10
+        );
+        const secondMintResult = await lazyMintAsset(
+          secondMintSignature,
+          4,
+          5,
+          metadataHashes[0],
+          10,
+          creator.address
+        );
+        const secondEventData = secondMintResult.events.filter(
+          (e: Event) => e.event == 'AssetLazyMinted'
+        )[0].args;
+
+        // tokenIds should be the same
+        expect(secondEventData.tokenId).to.equal(firstEventData.tokenId);
+
+        // users balance should be 10
+        expect(
+          await AssetContract.balanceOf(creator.address, firstEventData.tokenId)
+        ).to.equal(10);
+      });
+    });
+    describe('Revert', function () {
+      it('should revert if signature is invalid', async function () {
+        const {lazyMintAsset, metadataHashes, creator} =
+          await runCreateTestSetup();
+        const signature =
+          '0x45956f9a4b3f24fcc1a7c1a64f5fe7d21c00dd224a44f868ad8a67fd7b7cf6601e3a69a6a78a6a74377dddd1fa8c0c0f64b766d4a75842c1653b2a1a76c3a0ce1c';
+        await expect(
+          lazyMintAsset(signature, 4, 1, metadataHashes[0], 10, creator.address)
+        ).to.be.revertedWith('AssetCreate: Invalid signature');
+      });
+      it('should revert if the signature has been used before', async function () {
```

Comment:
if the nonce has been used

Reply:
Well, it's about reusing the signature, aka passing the same sig into the function again, but underneath it indeed is testing if the noce has been incremented correctly. 
I'd argue that this description is more accurate, happy to discuss this further though.

Reply:
Ok!